components/p3a/brave_p3a_service.cc

```diff
@@ -0,0 +1,362 @@
+
+void BraveP3AService::RegisterPrefs(PrefRegistrySimple* registry) {
+  BraveP3ALogStore::RegisterPrefs(registry);
+  registry->RegisterTimePref(kLastRotationTimeStampPref, {});
+  registry->RegisterBooleanPref(kP3AEnabled, true);
+
+  // New users are shown the P3A notice via the welcome page.
+  registry->RegisterBooleanPref(kP3ANoticeAcknowledged,
+                                first_run::IsChromeFirstRun());
+}
+
+void BraveP3AService::InitCallbacks() {
+  for (const char* histogram_name : kCollectedHistograms) {
+    base::StatisticsRecorder::SetCallback(
+        histogram_name,
+        base::BindRepeating(&BraveP3AService::OnHistogramChanged, this,
+                            histogram_name));
+  }
+}
+
+void BraveP3AService::Init() {
+  DCHECK(g_browser_process);
+
+  // Init basic prefs.
+  initialized_ = true;
+
+  average_upload_interval_ =
+      base::TimeDelta::FromSeconds(kDefaultUploadIntervalSeconds);
+
+  upload_server_url_ = GURL(kDefaultUploadServerUrl);
+  MaybeOverrideSettingsFromCommandLine();
+
+  VLOG(2) << "BraveP3AService::Init() Done!";
+  VLOG(2) << "BraveP3AService parameters are:"
+          << " upload_enabled_ = " << upload_enabled_
+          << ", average_upload_interval_ = " << average_upload_interval_
+          << ", randomize_upload_interval_ = " << randomize_upload_interval_
+          << ", upload_server_url_ = " << upload_server_url_.spec()
+          << ", rotation_interval_ = " << rotation_interval_;
+
+  InitPyxisMeta();
+
+  // Init log store.
+  log_store_.reset(
+      new BraveP3ALogStore(this, g_browser_process->local_state()));
+  log_store_->LoadPersistedUnsentLogs();
+  // Store values that were recorded between calling constructor and |Init()|.
+  for (const auto& entry : histogram_values_) {
+    log_store_->UpdateValue(entry.first.as_string(), entry.second);
+  }
+  histogram_values_ = {};
+  // Do rotation if needed.
+  const base::Time last_rotation =
+      local_state_->GetTime(kLastRotationTimeStampPref);
+  const base::TimeDelta last_rotation_interval =
+      rotation_interval_.is_zero() ? TimeDeltaTillMonday(last_rotation)
+                                   : rotation_interval_;
+  if (base::Time::Now() - last_rotation > last_rotation_interval) {
+    DoRotation();
+  }
+
+  // Init other components.
+  uploader_.reset(new BraveP3AUploader(
+      g_browser_process->shared_url_loader_factory(), upload_server_url_,
+      base::Bind(&BraveP3AService::OnLogUploadComplete, this)));
+
+  upload_scheduler_.reset(new BraveP3AScheduler(
+      base::Bind(&BraveP3AService::StartScheduledUpload, this),
+      (randomize_upload_interval_
+           ? base::BindRepeating(GetRandomizedUploadInterval,
+                                 average_upload_interval_)
+           : base::BindRepeating([](base::TimeDelta x) { return x; },
+                                 average_upload_interval_))));
+
+  // Start the engine if we are enabled.
+  if (upload_enabled_) {
```

Comment:
@bsclifton is the UI onboarding stuff we discussed in a separate PR? This appears to only be enabled by command line

Reply:
No, the onboarding is included here. It happens in brave://welcome. However, it only links to the brave://settings/privacy area where the toggle would need to be flipped. The infobar shown for existing users gives a one click disable